### PR TITLE
Show "Last Seen" in recent patients list

### PR DESCRIPTION
**Pull request by rakshakhegde**
https://www.pivotaltracker.com/story/show/166267273

### CODE REVIEW

**Comment by sanchita** [comment]
> https://www.pivotaltracker.com/story/show/166267273

Can you add more description to this story?
I do not understand what needs to be done here.
Also the tasks added are not descriptive.

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 1]
```diff
@@ -1093,43 +1093,6 @@ class PatientRepositoryAndroidTest {
     verifyRecentPatientOrder(patientWithBp2, facilityUuid = facility2Uuid)
```
Comment (sanchita) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt]: Please amend the first commit itself to fix this broken test.

File: app/src/main/java/org/simple/clinic/recentpatient/RecentPatientScreenItemTypes.kt [hunk 2]
```diff
@@ -57,47 +45,9 @@ data class RecentPatientItem(
...
-      val systolic: Int,
-      val diastolic: Int,
-      val updatedAtRelativeTimestamp: org.simple.clinic.summary.RelativeTimestamp
-  )
-
   class RecentPatientItemViewHolder(rootView: View) : ViewHolder(rootView) {
     val title by bindView<TextView>(R.id.recentpatient_item_title)
-    val lastBp by bindView<TextView>(R.id.recentpatient_item_last_bp)
-    val lastBpLabel by bindView<View>(R.id.recentpatient_item_last_bp_label)
+    val lastSeen by bindView<TextView>(R.id.recentpatient_item_last_seen)
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/recentpatient/RecentPatientScreenItemTypes.kt]: done

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -32,7 +32,7 @@
   <string name="patientsearchresults_register_patient_rationale_for_empty_state">Patient is not registered</string>
   <string name="patientsearchresults_register_patient">Register as a new patient</string>
   <string name="patientsearchresults_register_patient_for_empty_state">Register patient</string>
-  <string name="patientsearchresults_item_last_bp">Last BP:</string>
+  <string name="patientsearchresults_item_last_bp">Last Seen:</string>
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/res/values/strings.xml]: done

File: app/src/main/java/org/simple/clinic/recentpatient/RecentPatientScreenItemTypes.kt [hunk 4]
```diff
@@ -30,25 +23,20 @@ data class RecentPatientItem(
...
 
   @SuppressLint("SetTextI18n")
   override fun bind(viewHolder: RecentPatientItemViewHolder, position: Int) {
     viewHolder.apply {
       title.text = "$name, $age"
 
-      val lastBpText = lastBpText(itemView.context)
-      lastBp.text = lastBpText
-      lastBpLabel.visibleOrGone(lastBpText.isNotBlank())
+      lastSeen.text = " " + updatedAt.displayText(itemView.context)
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/recentpatient/RecentPatientScreenItemTypes.kt]: done

**Comment by rakshakhegde** [comment]
PRD for defining an "encounter": https://docs.google.com/document/d/1-0q1yRnkFHmEAdkpgaBK7hVHSgEfo_55LfwseFWDzSw

File: app/src/main/java/org/simple/clinic/recentpatient/RecentPatientScreenItemTypes.kt [hunk 5]
```diff
@@ -10,16 +9,10 @@ import io.reactivex.subjects.Subject
 import kotterknife.bindView
 import org.simple.clinic.R
 import org.simple.clinic.patient.Gender
-import org.simple.clinic.recentpatient.DateHeader.DateHeaderViewHolder
 import org.simple.clinic.recentpatient.RecentPatientItem.RecentPatientItemViewHolder
-import org.simple.clinic.recentpatient.RelativeTimestamp.OlderThanTwoDays
-import org.simple.clinic.recentpatient.RelativeTimestamp.Today
-import org.simple.clinic.recentpatient.RelativeTimestamp.Yesterday
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/recentpatient/RecentPatientScreenItemTypes.kt]: done

File: app/src/main/java/org/simple/clinic/recentpatient/RecentPatientScreenItemTypes.kt [hunk 6]
```diff
@@ -30,25 +23,20 @@ data class RecentPatientItem(
     val uuid: UUID,
     val name: String,
     val age: Int,
-    val lastBp: LastBp?,
     val gender: Gender,
-    val updatedAt: Instant
+    val updatedAt: RelativeTimestamp
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/recentpatient/RecentPatientScreenItemTypes.kt]: Renamed to `lastSeenTimestamp` as it'll be more in sync with feature we're building.

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt (hunk 1 above)
Comment (rakshakhegde) [reply app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt]: done

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -32,7 +32,7 @@
   <string name="patientsearchresults_register_patient_rationale_for_empty_state">Patient is not registered</string>
   <string name="patientsearchresults_register_patient">Register as a new patient</string>
   <string name="patientsearchresults_register_patient_for_empty_state">Register patient</string>
-  <string name="patientsearchresults_item_last_bp">Last BP:</string>
+  <string name="patientsearchresults_item_last_bp">Last Seen:\u0020</string>
```
Comment (sanchita) [inline app/src/main/res/values/strings.xml]: Please create a new string for `recent_patient` and revert the string for `patientsearchresults`.

Comment (sanchita) [reply app/src/main/res/values/strings.xml]: This has created a bug where `LAST BP` is now called `LAST SEEN` in Patient search results.

Comment (rakshakhegde) [reply app/src/main/res/values/strings.xml]: resolved

**Pull request merged**